### ISSUE TITLE
UTF-8 without ICONV returns wrong byte sequences

### ISSUE THREAD

**Issue by marsupilami79** [FreeTDS/freetds#49]
Hello,

I compiled FreeTDS using Cmake and Visual Studio 2013. For testing I generated sequences of random characters. One of those sequences is: Ē洅乕ཥ깳몊䚣ഒ컭閥

I used SQL Server Management Studio on SQL Express 2008R2 to insert this sequence. When fetching it from the database and setting freetds to use UTF-8 by calling dbSetLCharSet(LoginRec, "UTF-8") a freetds with ICONV support returns the following correct byte sequence:
C4 92 E6 B4 85 E4 B9 95 E0 BD A5 EA B9 B3 EB AA 8A E4 9A A3 E0 B4 92 EC BB AD E9 96 A5
If I use a FreeTDS without ICONV support, I get the following incorrect byte sequence:
C4 92 E6 B4 85 E4 B9 95 E1 FD A5 EA B9 B3 EB AA 8A E4 9A A3 E6 F4 92 EC BB AD E9 96 A5

I can only assume that the iconv.c in the replacements is somehow faulty.

**Comment by freddy77** [comment]
Which FreeTDS version are you using?
I think you are using DBSETLCHARSET with db library.
Can you post a TDSDUMP of your program? See http://www.freetds.org/userguide/logging.htm.
I looked at the code and I cannot find any way to have this result.

**Comment by marsupilami79** [comment]
I am using the R0_95 version, downloaded from Github some days ago. I have to check for the exact date. I am building FreeTDS on Windows with Visual C++ 2013. The project files are generated with cmake-gui. The FreeTDS with ICONV is built using liiconv 1.14, I tink - the latest version that is available from Savannah.

We do use FreeTDS with the dblib interface.

For the test case I first created 1000 tables with names consisting of 10 random Unicode characters selected from the Uniode Character Database in the XML version. Then I do a `SELECT TABLE_NAME FROM INFORMATION_SCHEMA.TABLES order by 1`.

The FreeTDS with libiconv returned the following bytes for the table name in question:
C4 92 E6 B4 85 E4 B9 95 E0 BD A5 EA B9 B3 EB AA 8A E4 9A A3 E0 B4 92 EC BB AD E9 96 A5
The FreeTDS without libiconv returned the following bytes:
C4 92 E6 B4 85 E4 B9 95 E1 FD A5 EA B9 B3 EB AA 8A E4 9A A3 E6 F4 92 EC BB AD E9 96 A5

I tried to reduce the size of the test case and created a database with only one table. To my surprise FreeTDS without libiconv suport returned a different byte sequence than in the first test case:
C4 92 E6 B4 85 E4 B9 95 00 FD A5 EA B9 B3 EB AA 8A E4 9A A3 00 F4 92 EC BB AD E9 96 A5 

For your reference I added my 1000 table names sql script and the table names in a zip file. Also I added log files for the last, reduced test case.


[tables.zip](https://github.com/FreeTDS/freetds/files/120926/tables.zip)
[logs.zip](https://github.com/FreeTDS/freetds/files/120927/logs.zip)
